server/jetstream_cluster_test.go

```diff
@@ -7640,6 +7640,59 @@ func TestJetStreamClusterMaxConsumers(t *testing.T) {
 	}
 }
 
+func TestJetStreamClusterMaxConsumersMultipleConcurrentRequests(t *testing.T) {
+	c := createJetStreamClusterExplicit(t, "JSC", 3)
+	defer c.shutdown()
+
+	nc, js := jsClientConnect(t, c.randomServer())
+	defer nc.Close()
+
+	cfg := &nats.StreamConfig{
+		Name:         "MAXCC",
+		Storage:      nats.MemoryStorage,
+		Subjects:     []string{"in.maxcc.>"},
+		MaxConsumers: 1,
+		Replicas:     3,
+	}
+	if _, err := js.AddStream(cfg); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	si, err := js.StreamInfo("MAXCC")
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if si.Config.MaxConsumers != 1 {
+		t.Fatalf("Expected max of 1, got %d", si.Config.MaxConsumers)
+	}
+
+	startCh := make(chan bool)
+	var wg sync.WaitGroup
+
+	for n := 0; n < 10; n++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			nc, js := jsClientConnect(t, c.randomServer())
+			defer nc.Close()
+			<-startCh
+			js.SubscribeSync("in.maxcc.foo")
```

Comment:
have to be careful that this doesnt unsubscrie (nc.Close()) before the other go routine is going on, my sample test had a wait on a channel there to prevent the defers from firing

Reply:
Did not affect my tests, was getting failure everytime.

Reply:
ok :)